File: RoundNumber.cpp
#include <bits/stdc++.h>
using namespace std;

int main(void) {
    int t;
    cin >> t;
    while (t--) {
        int n;
        cin >> n;
        vector<int> arr;
        int power = 1;
        while (n) {
            if (n%10)
                arr.push_back(n%10*power);
            n /= 10;
            power *= 10;
        }
        cout << arr.size() << endl;
        for (auto &x:arr)
            cout << x << ' ';
        cout << endl;
    }

    return 0;
}

File: Python/RoundNumber.py
def main():
    t = int(input())
    while t:
        n = int(input())
        arr = []
        power = 1
        while n:
            if n % 10:
                arr.append(n % 10*power)
            n /= 10
            power *= 10
        print(len(arr))
        for i in arr:
            print(i, end=" ")
        print()
        t -= 1


if __name__ == '__main__':
    main()
